sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/AnnotationProcessor.java

```diff
@@ -168,6 +168,17 @@ private HttpRequestContext createHttpRequestContext(ExecutableElement requestMet
         method.setExpectedStatusCodes(httpRequestInfo.expectedStatusCodes());
         method.addStaticHeaders(httpRequestInfo.headers());
         method.addStaticQueryParams(httpRequestInfo.queryParams());
+        TypeMirror returnValueWireType = null;
+        try {
+            // This will throw MirroredTypeException at compile time
+            // The assignment to clazz is only there to trigger the exception and use the TypeMirror from the exception.
+            Class<?> clazz = httpRequestInfo.returnValueWireType();
+        } catch (MirroredTypeException mte) {
+            TypeMirror typeMirror = mte.getTypeMirror();
```

Comment:
Is this the only way to access `TypeMirror`? It feels hacky.

Reply:
This is a common pattern in Java annotation processing to get the[TypeMirror of a class-valued annotation property. At compile time, you can't directly access the Class<?> value (it would cause a MirroredTypeException, so you intentionally trigger the exception and then extract the TypeMirror from it.
The [official Java documentation for MirroredTypeException](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/type/MirroredTypeException.html) describes this pattern

Reply:
I didn't find anything about the pattern in the documentation link you shared, but it makes sense to keep this as is based on what you said.